eng/pipelines/eng-common-sync.yml

```diff
@@ -30,16 +30,42 @@ stages:
         displayName: Sync ${{ parameters.DirectoryToSync }} Directory
 
         steps:
+        - ${{ if eq(variables['Build.Reason'], 'PullRequest') }}:
+          - pwsh: |
+              Set-PsDebug -Trace 1
+              $patchDir = "$(Build.ArtifactStagingDirectory)/patchfiles"
+              New-Item $patchDir -Type Directory
+              $patchfiles = git format-patch HEAD...origin/$(system.pullRequest.targetBranch) -- "${{ parameters.DirectoryToSync }}"
```

Comment:
From my local testing you don't need to create the patch directory nor do you have to move them. You can definitely use the --output-directory. Here is the command that I was able to use.

```
$patchfiles = git format-patch --output-directory $patcheDir HEAD...origin/$(system.pullRequest.targetBranch) -- "${{ parameters.DirectoryToSync }}"
```